File: frontend/src/types/github.ts
export interface LanguageMetric {
  language: string;
  count: number;
  percentage: number;
}

export interface ActivityMetrics {
  commits: number;
  prs: number;
  issues: number;
}

export interface RepositorySummary {
  name: string;
  description: string | null;
  html_url: string;
  stargazers_count: number;
  forks_count: number;
  language: string | null;
  updated_at: string;
}

export interface GithubSummary {
  username: string;
  name: string;
  avatar_url: string | null;
  bio: string | null;
  company: string | null;
  location: string | null;
  html_url: string;
  public_repos: number;
  followers: number;
  following: number;
  total_stars: number;
  total_forks: number;
  languages: LanguageMetric[];
  recent_activity: ActivityMetrics;
  activity_score: number;
  developer_tier: string;
  top_repositories: RepositorySummary[];
  profile_improvements: string[];
}

File: backend/app/schemas/github.py
from pydantic import BaseModel


class LanguageMetric(BaseModel):
    language: str
    count: int
    percentage: float


class ActivityMetrics(BaseModel):
    commits: int
    prs: int
    issues: int


class RepositorySummary(BaseModel):
    name: str
    description: str | None = None
    html_url: str
    stargazers_count: int
    forks_count: int
    language: str | None = None
    updated_at: str


class GithubSummaryResponse(BaseModel):
    username: str
    name: str
    avatar_url: str | None = None
    bio: str | None = None
    company: str | None = None
    location: str | None = None
    html_url: str
    public_repos: int
    followers: int
    following: int
    total_stars: int
    total_forks: int
    languages: list[LanguageMetric]
    recent_activity: ActivityMetrics
    activity_score: int
    developer_tier: str
    top_repositories: list[RepositorySummary]
    profile_improvements: list[str]

